packages/client/hmi-client/src/components/model/petrinet/tera-other-concepts-table.vue

```diff
@@ -0,0 +1,410 @@
+								target="_blank"
+								rel="noopener noreferrer"
+								:href="getCurieUrl(data.suggestedValue?.grounding_id)"
+								aria-label="Open Concept"
+							>
+								<i class="pi pi-external-link" />
+							</a>
+						</span>
+						<span v-else>
+							{{ data.suggestedValue }}
+						</span>
+					</template>
+				</Column>
+				<Column field="currentValue">
+					<template #header>
+						Current values for:
+						<AutoComplete
+							v-model="semanticSearchTerm"
+							placeholder="Select a variable or parameter"
+							:suggestions="filteredSemantics"
+							@complete="onSearch"
+							@item-select="onItemSelect"
+							optionLabel="id"
+							optionGroupLabel="label"
+							optionGroupChildren="items"
+							dropdown
+						/>
+					</template>
+					<template #body="{ data }">
+						<span v-if="data.key === ComparisonKey.GROUNDING && data.currentValue?.identifiers">
+							{{
+								getNameOfCurieCached(
+									nameOfCurieCache,
+									getCurieFromGroudingIdentifier(data.currentValue.identifiers)
+								)
+							}}
+							<a
+								v-if="data.currentValue"
+								target="_blank"
+								rel="noopener noreferrer"
+								:href="getCurieUrl(getCurieFromGroudingIdentifier(data.currentValue.identifiers))"
+								aria-label="Open Concept"
+							>
+								<i class="pi pi-external-link" />
+							</a>
+						</span>
+						<span v-else>
+							{{ data.currentValue }}
+						</span>
+					</template>
+				</Column>
+			</DataTable>
+			<template #footer>
+				<Button
+					:disabled="isEmpty(selectedAssignmentRows) || isEmpty(selectedCurrentSemantic)"
+					label="Replace current with selected"
+					@click="updateModel"
+				/>
+				<Button outlined label="Cancel" @click="onCloseAssignemntModal" />
+			</template>
+		</tera-modal>
+	</Teleport>
+</template>
+
+<script setup lang="ts">
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+import { Initial, Model, ModelParameter } from '@/types/Types';
+import { computed, ref } from 'vue';
+import { cloneDeep, groupBy, isEmpty } from 'lodash';
+import { Dictionary } from 'vue-gtag';
+import {
+	getCurieUrl,
+	parseCurie,
+	getCurieFromGroudingIdentifier,
+	getNameOfCurieCached
+} from '@/services/concept';
+import Button from 'primevue/button';
+import TeraModal from '@/components/widgets/tera-modal.vue';
+import AutoComplete, {
+	AutoCompleteCompleteEvent,
+	AutoCompleteItemSelectEvent
+} from 'primevue/autocomplete';
+import { isInitial, isModelParameter } from '@/utils/data-util';
+
+const props = defineProps<{
+	model: Model;
+}>();
+
+enum ComparisonKey {
+	NAME = 'name',
+	DESCRIPTION = 'description',
+	GROUNDING = 'grounding',
+	VALUE = 'value'
+}
+interface ComparisonData {
+	key: ComparisonKey;
+	name: string;
+	suggestedValue: any;
+	currentValue: any;
+}
+const emit = defineEmits(['update-model']);
+
+const isAssignmentModalVisible = ref(false);
+
+const nameOfCurieCache = ref(new Map<string, string>());
```

Comment:
I have seen this cache construction a couple of times before, would it make sense to cache at the application level instead of at the component level?

Reply:
I agree, I wanted to add caching to the API endpoints on the `hmi-server`, but... you know... evaluation time.